dev/WindowsAppRuntime_BootstrapDLL/MddBootstrapTracelogging.cpp

```diff
@@ -6,13 +6,13 @@
 #include "pch.h"
 
 void MddBootstrap_StopActivity(
-    const std::string failureType,
+    const std::string& failureType,
     const bool isActivityRunning,
     const GUID *activityId,
     const WindowsAppRuntime::MddBootstrap::Activity::Context& activityContext,
     const wil::FailureInfo& failure)
 {
-    MddBootstrap_WriteEventWithActivity(*failureType.c_str(), activityId);
+    MddBootstrap_WriteEventWithActivity(failureType.c_str(), activityId);
```

Comment:
AIUI this is still incorrect: the inner ETW macro is documented as requiring a string *literal*, and uses sizeof to get the string length. With this code, it will get the size of a pointer instead:

```C
int main() {
  const char* str = "1234567890123";
  printf("%d %d\n", sizeof("1234567890123"), sizeof(str));
}
```

Output:

```
Execution build compiler returned: 0
Program returned: 0
14 8
```

https://godbolt.org/z/eGG8n794d

Reply:
I don't think there's a way to fix this without substantially changing this function  (e.g. replacing it with a macro), as ETW does not support variable event names.

Reply:
Yeah, my local testing is pointing that out (I threw up the PR to get it backed up and optimistically start a pipeline build while I do local build+test). The macro definitions steep several layers deep but string literal's a necessity. Juggling code t comply...